client/state/shortcodes/reducer.js

```diff
@@ -46,6 +51,36 @@ export const requesting = createReducer(
 	}
 );
 
+function mediaItemsReducer( state, { siteId, data } ) {
+	if ( ! state.hasOwnProperty( siteId ) ) {
+		return state;
+	}
+
+	if ( ! ( data && data.media ) ) {
+		return state;
+	}
+	const media = castArray( data.media );
+	const updatedIds = media.map( item => String( item.ID ) );
+
+	return {
+		...state,
+		[ siteId ]: pickBy( state[ siteId ], shortcode => {
```

Comment:
This reducer will always mutate the state even if no matching shortcode is found. To get an A grade, it should return unmodified state.

Reply:
All reducers in this module would benefit from using `keyedReducer`. Potential future improvement.

Reply:
yup, future pr